### PR TITLE
Time in notification is now is sending time

### LINKED ISSUE #663: Notification time should be event time
Currently, the time that is shown in the notifications is the time when the notification was received by the device not the actual event time.

How to reproduce:
Set device to flight mode, send notification, wait a minute, go back online.

It should be be enough to add a `.setWhen(time_in_millis)` in GcmIntentService. Event time should be part of the message already.

### PR DESCRIPTION
**Pull request by redeye86**
Ensures that notifications show the actual event/sending time instead of the reception time on the device when you have a weak signal and receive the messages late.

Fixes #663
AFAIK the the google.sent_time is set when the cloud service sends the message to google, so technically it is not the actual event time, but

### CODE REVIEW

**Comment by mueller-ma** [comment]
Is google.sent_time always provided or do we need any fallback?

File: mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java [hunk 1]
```diff
@@ -65,8 +64,20 @@ protected void onHandleIntent(Intent intent) {
                     notificationId = Integer.parseInt(intent.getExtras().getString(EXTRA_NOTIFICATION_ID));
                 }
                 if ("notification".equals(intent.getExtras().getString("type"))) {
-                    sendNotification(intent.getExtras().getString(EXTRA_MSG), intent.getExtras().getLong(EXTRA_SENT_TIME), notificationId);
-                // If this is hideNotification, cancel existing notification with it's id
+                    //we use local time as fallback
+                    long timestamp = System.currentTimeMillis();
+                    if(intent.getExtras().containsKey("timestamp")){
```
Comment (mueller-ma) [inline mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java]: Is there an open issue on openhab-cloud for timestamp?

File: mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java [hunk 2]
```diff
@@ -33,7 +33,6 @@
     private static final String TAG = GcmIntentService.class.getSimpleName();
 
     public static final String EXTRA_MSG = "message";
-    public static final String EXTRA_SENT_TIME = "google.sent_time";
```
Comment (mueller-ma) [inline mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java]: Can you use constants for both keys?

**Comment by redeye86** [comment]
Ok, 

seems that i have messed it up. I accidentally merged my old attempt to implement home screen widgets. I will try to fix it....

**Comment by redeye86** [comment]
ok, i made a new fork so that we have a clean PR without all the unrelated commits. Please ignore this pr and use #685 instead!

**Pull request closed without merging**